src/main/java/net/mcreator/ui/modgui/codeviewer/ModElementCodeViewer.java

```diff
@@ -74,19 +97,37 @@ private void reload() {
 				try {
 					List<GeneratorFile> files = modElementGUI.getModElement().getGenerator()
 							.generateElement(modElementGUI.getElementFromGUI(), false, false);
+					files.sort(Comparator.<GeneratorFile, String>comparing(
+									e -> FilenameUtils.getExtension(e.file().getName()))
+							.thenComparing(e -> e.file().getName()));
 
-					files.sort(
-							Comparator.comparing(e -> FilenameUtils.getExtension(((GeneratorFile) e).file().getName()))
-									.thenComparing(e -> ((GeneratorFile) e).file().getName()));
+					List<GeneratorTemplatesList> lists = modElementGUI.getModElement().getGenerator()
+							.getModElementListTemplates(modElementGUI.getModElement(),
+									modElementGUI.getElementFromGUI());
 
 					for (GeneratorFile file : files) {
+						Optional<GeneratorTemplatesList> ownerListOptional = lists.stream()

```

Comment:
Is using `isGeneratedFromListTemplate` here needed, can't you check if `GeneratorFile file` is instanceof ListTemplate and in this case, use `getTemplatesList`?

Reply:
No, both times this method is currently called there's no `GeneratorTemplate` object that can be checked.

Reply:
I see. Maybe you could make GeneratorFile keep reference of GeneratorTemplate it was generated from so this lookup is not needed? It may come handy later to to keep this reference

Reply:
✔️ Fixed

Reply:
Looks like it's possible to avoid the remaining lookup happening upon removing outdated files. Just cache map keys need to be `GeneratorFile` instead of `File`. Is it a good idea?

Reply:
What cache map do you have in mind?

Reply:
The one in code preview generator class storing file code viewers.